config/src/main/scala/io/buoyant/config/Parser.scala

```diff
@@ -84,7 +84,7 @@ object Parser {
 
     // Subtypes must not conflict
     for (kinds <- configInitializers) {
-      ensureUniqueKinds(kinds)
+      kinds.groupBy(_.configClass.getSuperclass).values.foreach(ensureUniqueKinds)
```

Comment:
I think conflicting kinds between response classifiers for different protocols is going to be a problem at deserialization time because they are all subtypes of ResponseClassifierConfig.  So I think we want to keep this check.